internal/pkg/util/shell/interpreter/interpreter.go

```diff
@@ -57,6 +61,15 @@ func defaultExecHandler(ctx context.Context, args []string) error {
 		return interp.NewExitStatus(127)
 	}
 
+	timeoutVal := GetEnvVar(hc, runscriptTimeoutEnv)
```

Comment:
I wouldn't do that there, instead you can replace this code with:
```
func defaultExecHandler(ctx context.Context, args []string) error {
	hc := interp.HandlerCtx(ctx)
	path, err := interp.LookPath(hc.Env, args[0])
	if err != nil {
		fmt.Fprintln(hc.Stderr, err)
		return interp.NewExitStatus(127)
	}

    if ctx.Deadline() == nil {
        ectx, cancel := context.WithTimeout(ctx, execTimeout)
        defer cancel()
    } else {
        ectx = ctx
    }
```

Reply:
thanks! I've updated the code in this commit
https://github.com/apptainer/apptainer/pull/2116/commits/db81c41fb8117a6b696fe8d31a895c024b9135e8